report-service/src/main/java/gov/cdc/dataingestion/authservice/integration/service/AuthService.java

```diff
@@ -0,0 +1,180 @@
+package gov.cdc.dataingestion.authservice.integration.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import jakarta.annotation.PostConstruct;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.config.Registry;
+import org.apache.http.config.RegistryBuilder;
+import org.apache.http.conn.socket.ConnectionSocketFactory;
+import org.apache.http.conn.socket.PlainConnectionSocketFactory;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.BasicHttpClientConnectionManager;
+import org.apache.http.ssl.SSLContexts;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import javax.crypto.spec.SecretKeySpec;
+import javax.net.ssl.SSLContext;
+import java.security.Key;
+import java.util.Base64;
+
+@Service
+@EnableScheduling
+public class AuthService {
+    private static final Logger logger = LoggerFactory.getLogger(AuthService.class);
+
+    @Value("${auth.username}")
+    private String nbsUsername;
+
+    @Value("${auth.password}")
+    private String nbsPassword;
+
+    @Value("${auth.sign-on-url}")
+    private String signOnUrl;
+
+    @Value("${auth.token-url}")
+    private String tokenUrl;
+
+    @Value("${auth.salt-for-algorithm}")
+    private String saltForAlgorithm;
+
+    private String token;
+
+    private static final String AUTH_ELR_CLAIM = "ELR Importer";
+    private static final String AUTH_ECR_CLAIM = "ECR Importer";
+
+    private String encodedSignOnUrl;
+
+
+    @PostConstruct
+    public void generateAuthTokenDuringStartup() {
+        encodedSignOnUrl = getSignOnUrl();
+        generateToken(encodedSignOnUrl);
+
+        String authRoleName = getAuthRoleClaim(token);
+        if(authRoleName == null || authRoleName.isEmpty()) {
+            logger.error("Auth roles not defined, nothing to authorize.");
+            return;
+        }
+
+        logger.info("Auth role claim is: " + authRoleName);
```

Comment:
This could be updated to logger.debug. If you need it to be logger.info then please add a note somewhere.

Same thing with other logger.info in this pr

Reply:
Updated to debug logger